synapse/handlers/presence.py

```diff
@@ -880,29 +884,45 @@ async def _handle_state_delta(self, deltas):
                     # Ignore changes to join events.
                     continue
 
-            await self._on_user_joined_room(room_id, state_key)
+            # Retrieve any user presence state updates that need to be sent as a result,
+            # and the destinations that need to receive it
+            destinations, user_presence_states = await self._on_user_joined_room(
+                room_id, state_key
+            )
+
+            # Insert the destinations and respective updates into our destinations dict
+            for destination in destinations:
+                presence_destinations.setdefault(destination, set())
+                presence_destinations[destination].update(user_presence_states)
```

Comment:
Why is this on two lines, rather than `.setdefault(..).update(..)`?

Reply:
Awkwardly it doesn't fit within 95 characters on one line, but I can still combine it into one contiguous statement.

Reply:
The second lookup in the dict, `presence_destinations[destination]`, was making me sad :)